presto-hive/src/main/java/io/prestosql/plugin/hive/HiveColumnHandle.java

```diff
@@ -65,47 +68,98 @@
         SYNTHESIZED,
     }
 
+    private final String baseColumnName;
+    private final int baseHiveColumnIndex;
+    private final HiveType baseColumnHiveType;
+    private final Type baseColumnType;
+
+    private final List<String> dereferenceFieldNames;
+    private final List<Integer> dereferenceFieldIndices;
+
     private final String name;
+
     private final HiveType hiveType;
     private final Type type;
-    private final int hiveColumnIndex;
+
     private final ColumnType columnType;
     private final Optional<String> comment;
 
     @JsonCreator
     public HiveColumnHandle(
-            @JsonProperty("name") String name,
+            @JsonProperty("baseColumnName") String baseColumnName,
+            @JsonProperty("baseHiveColumnIndex") int baseHiveColumnIndex,
+            @JsonProperty("baseColumnHiveType") HiveType baseColumnHiveType,
+            @JsonProperty("baseColumnType") Type baseColumnType,
```

Comment:
Yes, I'll add some comments. Was also thinking that we can move the `hiveType` and `type` out of the constructor arguments, since they can be derived from other info during the construction. what do you think?